File: Problems/0388-longest-absolute-file-path.py
class Solution(object):
    def lengthLongestPath(self, input):
        """
        :type input: str
        :rtype: int
        """
        depth_len = {0: 0}
        ans = 0

        for s in input.split('\n'):
            depth = s.count('\t')
            name = s[depth:]

            if '.' in name:
                ans = max(ans, depth_len[depth] + len(name))
            else:
                depth_len[depth + 1] = depth_len[depth] + len(name) + 1

        return ans

File: Problems/0388-longest-absolute-file-path.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here

class Solution {
public:
    int lengthLongestPath(string input) {
        unordered_map<int, int> depth_len;
        depth_len[0] = 0;
        int ans = 0;

        stringstream ss(input);
        string s;

        while (getline(ss, s, '\n')) {
            int depth = 0;

            while (depth < s.size() && s[depth] == '\t')
                depth++;

            string name = s.substr(depth);

            if (name.find('.') != string::npos) {
                ans = max(ans, depth_len[depth] + (int)name.size());
            } else {
                depth_len[depth + 1] =
                    depth_len[depth] + (int)name.size() + 1;
            }
        }

        return ans;
    }
};
